api4/post.go

```diff
@@ -316,6 +338,54 @@ func searchPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(c.App.PostListWithProxyAddedToImageURLs(posts).ToJson()))
 }
 
+func hasTeamOrSystemContextPermission(teamID string, userID string, permission *model.Permission, app *app.App) bool {
```

Comment:
Might be worth throwing this function and the two below into the `app` package (intersect() would probably be better in `utils` actually). That way we can keep the api packages lighter and I can definitely see potential for these functions to be re-used by other packages in the future, similar to the other permission related functions